crates/tuono/src/build.rs

```
@@ -0,0 +1,89 @@
+use console::Term;
+use fs_extra::dir::{copy, CopyOptions};
+use spinners::{Spinner, Spinners};
+use std::io;
+use std::path::PathBuf;
+use std::thread::sleep;
+use std::time::Duration;
+
+use crate::app::App;
+use crate::mode::Mode;
+
+#[tokio::main]
+pub async fn build(mut app: App, ssg: bool, no_js_emit: bool) -> Result<(), io::Error> {
+    if no_js_emit {
+        println!("Rust build successfully finished");
+        return Ok(());
+    }
+
+    if ssg && app.has_dynamic_routes() {
+        // TODO: allow dynamic routes static generation
+        println!("Cannot statically build dynamic routes");
+        return Ok(());
+    }
+
+    app.build_tuono_config()
+        .expect("Failed to build tuono.config.ts");
+
+    // start showing the spinner
+    let term = Term::stdout();
```

I'd remove the Term interface to replace the spinner with a ✅ rather than removing the line below
```suggestion
```

---

Also I'd put this at the beginning of the function

---

This is causing test fail because the spinner text is now included in the `stderr` which is checked by the following tests:
- `it_fails_without_installed_build_config_script`
- `it_fails_without_installed_build_script`

These tests check errors that are coming from `App` `build_tuono_config` function which exit the process with an error status, so there is no way to remove, nor replace the spinner from the `build` function.

IMHO we should return these errors to the caller,
so they can be properly managed, e.g.: display custom error messages / hiding spinner / exit the process.

However this seems out of scope for this PR, so I suggest we simply move `app_build_spinner` declaration `app.check_server_availability(Mode::Prod);` and treat this as a separate task.

Let me know what do you think about this.

---

As per Discord discussion we are going ahead using the workaround.